### ISSUE TITLE
[ML] ONNX Runtime Semantic Element Classifier for ir-classification

### ISSUE THREAD

**Issue by oscharko** [oscharko-dev/workspace-dev#941]
## Summary

Integrate a small (~50M parameter) ONNX classification model into the element classification pipeline to improve `ScreenElementType` predictions for ambiguous Figma nodes, executed via `onnxruntime-node` with CoreML Execution Provider on Apple Silicon.

Ref: Epic #857 | Depends on: #940 (ML Asset Management Infrastructure)

---

## Motivation

The current classification in `src/parity/ir-classification.ts` uses declarative `ClassificationRule` objects evaluated by priority against `NodeClassificationContext` (line 85). This works well for clear-cut cases but struggles with ambiguous nodes — e.g., distinguishing a search input from a text field, or a rating widget from a button group. A supervised classifier trained on labeled Figma-to-MUI mappings can improve confidence on the 35+ `ScreenElementType` variants.

**Abgrenzung zu #937:** Issue #937 covers dense embeddings for *retrieval* (finding similar past corrections). This issue covers *supervised classification* of Figma nodes into MUI component types — different model architecture, different training objective, different integration point.

---

### Model Architecture

- **Input:** Feature vector derived from `NodeClassificationContext` (26 boolean context keys + numeric properties: fontSize, width, height, depth, childCount, gap, cornerRadius, padding)
- **Output:** Probability distribution over `ScreenElementType` variants (35+ classes)
- **Architecture:** Small feedforward network or shallow transformer (~50M parameters)
- **Format:** ONNX (exported via PyTorch `torch.onnx.export()` or Hugging Face Optimum)
- **Quantization:** INT8 (q8) for minimal quality loss, ~50 MB on disk

### Integration Point

**File:** `src/parity/ir-classification.ts`
**Location:** After `createNodeClassificationContext()` (line 1143), before `matchesNodeRule()` (line 1024)

```
Existing flow:
  createNodeClassificationContext() → matchesNodeRule() per ClassificationRule → best match

Enhanced flow (ml_inference mode only):
  createNodeClassificationContext() → MlElementClassifier.predict(context) → 
    if confidence > threshold: use ML prediction
    else: fall through to matchesNodeRule() deterministic rules
```

### Runtime

- **Execution Provider:** CoreML EP on Apple Silicon (Neural Engine acceleration)
- **Fallback:** CPU EP when CoreML unavailable
- **Loading:** Via MlAssetManager (#940) — lazy `import("onnxruntime-node")`
- **Gating:** Only active when `llmCodegenMode === "ml_inference"` AND model available

### Performance Targets

| Metric | Target |
|---|---|
| Per-element inference | <5 ms (CoreML EP) |
| Per-screen (100 elements) | <500 ms |
| Model load time (cold) | <3 seconds |
| RAM footprint | 100-300 MB |

---

## Node.js 22 Feasibility

- `onnxruntime-node` publishes prebuilt arm64 darwin binaries, compatible with Node.js 20+
- CoreML Execution Provider: Requires macOS 10.15+, optimized for Apple Neural Engine
- Package installed as `optionalDependency` via #940 infrastructure

---

## Risks

| Risk | Mitigation |
|---|---|
| Native arm64 binary (~15 MB) | `optionalDependency` — graceful degradation |
| Model quality on unseen designs | Confidence threshold gating; deterministic fallback |
| Training data availability | Bootstrap from existing `ClassificationRule` outputs as weak labels |

---

## arXiv References

- ONNX Runtime optimization: https://arxiv.org/abs/1903.08560
- Neural Architecture Search for edge classification: https://arxiv.org/abs/1807.11626

---

## Acceptance Criteria

- [ ] `MlElementClassifier` wraps `NodeClassificationContext` into ONNX input tensor
- [ ] Inference returns confidence scores per `ScreenElementType`
- [ ] Predictions override rules only when confidence exceeds configurable threshold
- [ ] Deterministic fallback when model unavailable or confidence low
- [ ] CoreML EP used on Apple Silicon, CPU EP as fallback
- [ ] Classification results persisted as `ml.classification.result` artifact
- [ ] No regression in existing deterministic classification tests
- [ ] RAM stays within 300 MB ceiling for classifier

---

## Dependencies

- **Requires:** #940 (ML Asset Management Infrastructure)
- **Complements:** #917 (Feature Discretizer — consumes same feature space)
- **Related:** #920 (Advisory Similarity Ranking — different approach, same goal)

**Comment by oscharko** [comment]
Consolidated into umbrella epic #857 for staged execution. Reopen when the feedback/ML track is actively funded and scheduled.
